core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCommits.java

```diff
@@ -206,6 +191,89 @@ public void testFailedCommit() throws Exception {
     Assertions.assertThat(manifests).as("Should contain 0 Avro manifest files").isEmpty();
   }
 
+  @Test
+  public void testCommitFailedBeforeChangeVersionHint() throws Exception {
+    // create v1-format table,for easy testing
+    Map<String,String> confMap = new HashMap<>();
+    confMap.put(TableProperties.FORMAT_VERSION,"1");
+    String newLocation = tableLocation+"v1"+File.separator;
+    Table v1Table = TABLES.create(SCHEMA, SPEC,confMap,newLocation);
```

Comment:
Why are we making a new table here instead of just using the class owned table? I think all the other tables in this file just use `table` from the base class which automatically sets up a table

Reply:
Because it's easier to create a V1 table, modify the metadata and then simulate the commit. I tried to commit the metadata the other way at first, but it was too much trouble and there were a lot of judgement conditions preventing me from committing. So I got lazy.

Reply:
I'm not sure what you are saying here, this is more code than would be required with just using `table`.

Reply:
```table``` is v2 format. Since I'm not very good at ICEBERG-API, at first I tried to simulate a commit, but in the end I couldn't pass the judgement logic in the commit method, and the commit failed. So I read the test cases of other catalogues and chose the most convenient way to simulate a commit.

Reply:
I don't know what you mean by "Judgement logic, why not just do

```java
    table.updateSchema().addColumn("n", Types.IntegerType.get()).commit();
```
It's done in several other places in this file

Reply:
Oh.. I really haven't tried this. Let me try.

Reply:
I tried it, and so far it doesn't look like I have a way to inject the proxy ``Spy(HadoopTableOperations)`` into ``table``, where the definition of Operations in ``BaseTable`` is ``Final``.